test/src/test/java/hudson/cli/DisablePluginCommandTest.java

```diff
@@ -0,0 +1,376 @@
+        assertThat(disablePluginsCLiCommand("mandatory-depender", "dependee"), succeeded());
+        assertPluginDisabled("dependee");
+        assertPluginDisabled("mandatory-depender");
+    }
+
+    /**
+     * Can disable a plugin without dependents plugins and Jenkins restart after it if -restart argument is passed.
+     */
+    @Test
+    @Issue("JENKINS-27177")
+    @WithPlugin("dependee-0.0.2.hpi")
+    public void restartAfterDisable() {
+        assumeNotWindows();
+        assertThat(disablePluginsCLiCommand("-restart", "dependee"), succeeded());
+        assertPluginDisabled("dependee");
+        assertJenkinsInQuietMode();
+    }
+
+    /**
+     * Can disable a plugin without dependents plugins and Jenkins doesn't restart after it if -restart is not passed.
+     */
+    @Test
+    @Issue("JENKINS-27177")
+    @WithPlugin("dependee-0.0.2.hpi")
+    public void notRestartAfterDisablePluginWithoutArgumentRestart() {
+        assertThat(disablePluginsCLiCommand("dependee"), succeeded());
+        assertPluginDisabled("dependee");
+        assertJenkinsNotInQuietMode();
+    }
+
+    /**
+     * A non-existing plugin returns with a {@link DisablePluginCommand#RETURN_CODE_NO_SUCH_PLUGIN} status code.
+     */
+    @Test
+    @Issue("JENKINS-27177")
+    @WithPlugin("dependee-0.0.2.hpi")
+    public void returnCodeDisableInvalidPlugin() {
+        assertThat(disablePluginsCLiCommand("wrongname"), failedWith(RETURN_CODE_NO_SUCH_PLUGIN));
+    }
+
+    /**
+     * A plugin already disabled returns 0 and jenkins doesn't restart even though you passed the -restart argument.
+     * @throws IOException See {@link PluginWrapper#disable()}.
+     */
+    @Test
+    @Issue("JENKINS-27177")
+    @WithPlugin("dependee-0.0.2.hpi")
+    public void disableAlreadyDisabledPluginNotRestart() throws IOException {
+        // Disable before the command call
+        disablePlugin("dependee");
+
+        assertPluginDisabled("dependee");
+        assertThat(disablePluginsCLiCommand("-restart", "dependee"), succeeded());
+        assertPluginDisabled("dependee");
+        assertJenkinsNotInQuietMode();
+    }
+
+    /**
+     * If some plugins are disabled, Jenkins will restart even though the status code isn't 0 (is 16).
+     */
+    @Test
+    @Issue("JENKINS-27177")
+    @WithPlugin({"variant.hpi", "depender-0.0.2.hpi", "mandatory-depender-0.0.2.hpi", "plugin-first.hpi", "dependee-0.0.2.hpi", })
+    public void restartAfterDisablePluginsAndErrors() {
+        assumeNotWindows();
+        assertThat(disablePluginsCLiCommand("-restart", "variant", "dependee", "depender", "plugin-first", "mandatory-depender"), failedWith(RETURN_CODE_NOT_DISABLED_DEPENDANTS));
+        assertPluginDisabled("variant");
+        assertPluginEnabled("dependee");
+        assertPluginDisabled("depender");
+        assertPluginDisabled("plugin-first");
+        assertPluginDisabled("mandatory-depender");
+        assertJenkinsInQuietMode(); // some plugins were disabled
```

Comment:
I don't understand this comment.

Reply:
changed, i wanted to say that Jenkins is restarting because at least one plugin was disabled.